eq_bot/game/window.py

```diff
@@ -16,19 +18,31 @@
 EVERQUEST_ROOT_FOLDER=get_config('game.root_folder').rstrip("\\")
 
 
-class EverQuestWindow:
-    def __init__(self):
+class EverQuestWindow(Thread):
+    def __init__(self, daemon: bool = False):
+        super().__init__(daemon=daemon)
         self.player = CurrentPlayer(
             name=get_config('player.name'),
             server=get_config('player.server'),
             guild=get_config('player.guild'))
 
+        self._queue = Queue()
+
         if get_config('player.autodetect'):
             if not self.player.name or not self.player.server:
                 self._lookup_current_player()
             if not self.player.guild:
                 self._lookup_current_guild()
 
+    # Run this as a daemon so the thread will be cleaned up if the process is destroyed
+    def run(self) -> None:
+        while True:
+            handler = self._queue.get(block=True)
```

Comment:
Really weird. I was getting a null exception here because the `action` was not being passed in as a lambda.
```suggestion
            handler = self._queue.get(block=True)
            if not callable(handler):
                print(
                    f'Received an action of type {type(handler)}, rather than a function. The action will be ignored.',
                    flush=True)
                continue
```

I could see print statements executing and everything. Then I added this.

After I added a `lambda:` in front of the following call in buff_manager.py:
self.handle_tell_message_async(tell_message)

And everything worked.

But after that.. If I remove the `lambda:` and/or my above suggestion it still works.

It seems like it shouldn't work if I remove the lambda, since we'd be first executing `handle_tell_message_async` and passing its return function into `handle_window_action` 🤔